test/common/testlib.py

```diff
@@ -1156,10 +1156,10 @@ def filterOutput(self, test, failed, output):
         # Otherwise pass through this command if it exists
         cmd = [ "tests-policy", testvm.DEFAULT_IMAGE ]
         try:
-            proc = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
```

Comment:
better as subprocess.Popen(..., encoding='utf-8')

but I'd say the problem is rather related to the 'output' here already being a unicode string instead of a bytestring.  which way do we really want this?

Reply:
The `encoding` argument is not yet understood by [python 2's subprocess](https://docs.python.org/2/library/subprocess.html), and I want to keep this bilingual. (In fact, current master still runs tests as python 2, see PR #9571). So indeed it's more robust to just use bytes on stdin and out and thus entirely ignore the locale. Hence the encoding into stdin and decoding of its stdout.

Reply:
more specifically: are we intending to treat the test output as a string that we will operate on, or is it just a blob that we are moving around between different things?  I lean toward 'blob' here...

our current troubles are apparently coming from using .decode('utf-8') at the call site to .filterOutput().  That change was introduced only recently, with #9512.

Reply:
That's a good point, thanks! I sent PR #9582 with an alternative implementation that just keeps everything as bytes.